packages/datadog-plugin-nats/test/test-setup.js

```diff
@@ -0,0 +1,224 @@
+'use strict'
+
+class NatsTestSetup {
+  async setup (module) {
+    this.module = module
+    this.nc = null
+    this.subscription = null
+    this.requestResponder = null
+    this.sc = module.StringCodec()
+    this.nc = await module.connect({
+      servers: ['127.0.0.1:4222'],
+      timeout: 5000
+    })
+
+    // Set up responder BEFORE any tests run to ensure it's outside instrumentation context
+    this.requestResponder = this.nc.subscribe('time.request')
+    ;(async () => {
+      for await (const msg of this.requestResponder) {
+        msg.respond(this.sc.encode(JSON.stringify({ time: new Date().toISOString() })))
+      }
+    })()
+
+    // Send a test message to verify responder is working
+    await new Promise(resolve => setTimeout(resolve, 200))
```

Comment:
No need for the `Promise` constructor here. Just use `node:timers/promises`.

Broadly though, since it doesn't look like subprocesses are being used, there's no need to wait or time anything. We should be able to do the appropriate things when they're ready, not after timeouts. Timeouts like this nearly always add flakiness.